packages/common/src/types/pivot.ts

```diff
@@ -6,39 +6,40 @@ export type PivotConfig = {
     columnOrder?: string[];
 };
 
-export interface PivotValue {
+type Value = {
     raw: unknown;
     formatted: string;
-}
+};
 
-export type PivotFieldValueType =
+type HeaderOrIndexType =
     | { type: FieldType.METRIC; fieldId?: undefined }
     | { type: FieldType.DIMENSION; fieldId: string };
 
-export type TitleFieldValue = null | {
-    type: FieldType.DIMENSION;
+export type PivotHeaderType = HeaderOrIndexType;
+export type PivotIndexType = HeaderOrIndexType;
+
+export type PivotValue =
```

Comment:
What's the benefit of combining these types?

Aren't the values in `headerValues`/`indexValues` always with `type: 'label'` and `dataValues` are `type: 'value'` ?

If so we could avoid a lot of `if (type === 'label')` calls later

Reply:
Or is this to distinguish between the dimension values and metric names

Reply:
Ah I just saw the changes to `pivotQueryResults` - I found this pretty confusing because `label` vs `value` is quite generic but one of these is quite specific, it's a metric reference.

Reply:
> Or is this to distinguish between the dimension values and metric names
yes, but yeah, improvements needed if it was not clear the first time! gonna give it more thinking in the follow up PRs